src/App.tsx

```diff
@@ -15,48 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goodsFromServer];
+
+  visibleGoods.sort((a, b) => {
+    if (sortType === SortType.ALPHABET) {
+      return a.localeCompare(b);
+    }
+
+    if (sortType === SortType.LENGTH) {
+      return a.length - b.length;
+    }
+
+    return 0;
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleReset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  const reorderedGoods = getReorderedGoods({ sortType, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn({
+            'button is-info': true,
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={() => setSortType(SortType.ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn({
+            'button is-success': true,
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+          onClick={() => setSortType(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn({
+            'button is-warning': true,
+            'is-light': isReversed !== true,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortType !== SortType.NONE || isReversed)
```

Comment:
Nitpick: Sometimes when `render` becomes overcrowded or the logical assesment is complicated we can move it to it's separate variable. For example:
before render you define the following:
```typescript
const isResetButtonVisible = sortType !== SortType.NONE || isReversed
```
and then in `render` you have:
```typescript
isResetButtonVisible && (
            <button
              type="button"
              className="button is-danger is-light"
              onClick={() => handleReset()}
            >
              Reset
            </button>
          )}
```

It definitely helps with readability, although it is not strictly required here. 
Just letting you know about different approaches :)